[Dev] OONI Run v2

aanorbel opened the pull request:
## Proposed Changes

  -  Internal refactoring to enable use of descriptor format in application core. https://github.com/ooni/probe-ios/commit/d0ffa8ea32b9ef79dcca70f76d58439e014d1485

aanorbel committed:
feat: internal refactoring to enable use of descriptor format in application core.

bassosimone commented on ooniprobe/View/RunTest/TestOverviewViewController.m:
```diff
@@ -37,19 +38,15 @@ - (void)viewDidLoad {
-    if (testSuite.getTestList.count <= 0) {
-        self.runButton.userInteractionEnabled = NO;
-        [self.runButton setTitleColor:[UIColor colorNamed:@"color_gray3"] forState:UIControlStateNormal];
-    }
```
Can you elaborate on the implication of removing this block of code?

aanorbel replied:
This code was used to check if at least one test is enabled, which would then disabled the runButton.

Since there is going to be no run button anymore, no need to have it.

bassosimone replied:
Gotcha!

bassosimone commented on ooniprobe/View/RunTest/TestOverviewViewController.m:
```diff
@@ -81,14 +78,14 @@ -(void)changeConstraints{
+//        [self.estimatedDetailLabel setText:
+//         [NSString stringWithFormat:@"%@ %@",
+//          testSuite.dataUsage,
+//          [LocalizationUtility getReadableRuntime:[testSuite getRuntime]]]];
```
Is this code that we still need to convert or code that we can safely get rid with?

aanorbel replied:
Thanks for pointing this out. I need to convert it to something else. Thats why its commented not deleted

bassosimone replied:
Thanks!

bassosimone commented on ooniprobe/View/RunTest/TestOverviewViewController.m:
```diff
@@ -99,8 +96,8 @@ -(void)reloadLastMeasurement{
-    [testSuite.testList removeAllObjects];
-    [testSuite getTestList];
+//    [testSuite.testList removeAllObjects];
+//    [testSuite getTestList];
```
Is this code that we still need to convert or code that we can safely get rid with?

aanorbel replied:
I expect this to come in place when i am workin on the overview screen. No need to have it right now. I will check  make some adjustments.

bassosimone replied:
Okay, thank you!

bassosimone commented on ooniprobe/View/RunTest/TestOverviewViewController.m:
```diff
@@ -133,7 +130,7 @@ -(IBAction)run:(id)sender{
     if ([[segue identifier] isEqualToString:@"toTestSettings"]){
         SettingsTableViewController *vc = (SettingsTableViewController * )segue.destinationViewController;
-        [vc setTestSuite:testSuite];
+        //[vc setTestSuite:testSuite];
```
Is this code that we still need to convert or code that we can safely get rid with?

aanorbel replied:
There is no time we need to go directly to the settings from the TestOverview page. I am still looking into this.

bassosimone replied:
Alright!

bassosimone reviewed (commented):
I mainly have questions about commented out code

aanorbel committed:
fix: update based on PR review

aanorbel committed:
feat (OONI Run v2): Update Dashboard view to match new design  (#556)

Fixes https://github.com/ooni/ooni.org/issues/1512


## Proposed Changes

  - Update storyboard to match designs
  - Update `DashboardTableViewCell` definition to fine-tune views
 
|.|.|
|-|-|
| ![Simulator Screenshot - iPhone 15 Pro Max - 2024-02-27 at 21 09
40](https://github.com/ooni/probe-ios/assets/17911892/8fab41df-703e-49c5-90c6-7921173460be)
| ![Simulator Screenshot - iPhone 15 Pro Max - 2024-02-27 at 21 09
52](https://github.com/ooni/probe-ios/assets/17911892/334fa805-37af-4027-9ea7-fbecf033d944)
|

aanorbel committed:
feat (OONI Run v2): update preference screens to match new designs (#563)

Fixes https://github.com/ooni/ooni.org/issues/1516

## Proposed Changes

  - Updated Preference view.
  - Remove cell separators.

|.|.|.|
|-|-|-|
| ![Simulator Screenshot - iPhone 15 Pro Max - 2024-06-17 at 12 16
22](https://github.com/ooni/probe-ios/assets/17911892/da283012-625f-4aae-a679-2c0e81240ef9)|
![Simulator Screenshot - iPhone 15 Pro Max - 2024-06-17 at 12 16
27](https://github.com/ooni/probe-ios/assets/17911892/65429962-8fe5-4274-ac4a-377b9db6c40e)
| ![Simulator Screenshot - iPhone 15 Pro Max - 2024-06-17 at 12 16
31](https://github.com/ooni/probe-ios/assets/17911892/6508f17d-cca2-4364-93c2-b6cdae21d443)
|
| ![Simulator Screenshot - iPhone 15 Pro Max - 2024-06-17 at 12 16
36](https://github.com/ooni/probe-ios/assets/17911892/861ae3db-3707-4442-a567-300983434b32)
| ![Simulator Screenshot - iPhone 15 Pro Max - 2024-06-17 at 12 16
42](https://github.com/ooni/probe-ios/assets/17911892/63c8f7a5-5ca2-4abc-acc4-3daa28881bea)
| ![Simulator Screenshot - iPhone 15 Pro Max - 2024-06-17 at 12 16
58](https://github.com/ooni/probe-ios/assets/17911892/2ec2163e-28c8-4b3b-bb0d-db4abd9bc07b)
|
| ![Simulator Screenshot - iPhone 15 Pro Max - 2024-06-17 at 12 17
04](https://github.com/ooni/probe-ios/assets/17911892/e152614c-d4bc-4115-9eea-54e284ed2254)
|. |.|
